### PR TITLE
Readability refactor of wavenet.py

### PR DESCRIPTION
**Pull request by lemonzi**
Organize and document wavenet.py.

### CODE REVIEW

**Comment by lemonzi** [comment]
I haven't merged from upstream yet or checked that it works, but how's this looking?

**Comment by ibab** [comment]
I like the changes, but they conflict quite heavily with recent changes on `master`. Sorry for that.
I'll try to fix issue #11, which might require more changes to `wavenet.py`, and then we can make sure to clean it up properly.

**Comment by lemonzi** [comment]
Yep, let's close the open issues first. I'll force-undo the moving functions around to ease the merge.

**Comment by lemonzi** [comment]
With all the merging and conflict-fixing this has grown quite big.... How's it now? I haven't tested a full train/generate run, how much does Travis cover?

File: train.py [hunk 1]
```diff
@@ -236,21 +239,21 @@ def main():
...
-    
+
     # Create network.
     net = WaveNet(args.batch_size,
-                  wavenet_params["quantization_steps"],
-                  wavenet_params["dilations"],
-                  wavenet_params["filter_width"],
-                  wavenet_params["residual_channels"],
-                  wavenet_params["dilation_channels"])
+                  quantization_steps=wavenet_params["quantization_steps"],
```
Comment (ibab) [inline train.py]: This is a good idea!

File: wavenet.py [hunk 2]
```diff
@@ -105,71 +127,91 @@ def _create_network(self, input_batch):
...
-            # We skip connections from the outputs of each layer, adding them
-            # all up here.
-            total = sum(outputs)
-            transformed1 = tf.nn.relu(total)
-            conv1 = tf.nn.conv1d(transformed1, w1, stride=1, padding="SAME")
-            transformed2 = tf.nn.relu(conv1)
-            conv2 = tf.nn.conv1d(transformed2, w2, stride=1, padding="SAME")
-
-        return conv2
+            layer_channels = [self.residual_channels,
```
Comment (ibab) [inline wavenet.py]: Great, this will help with making this part more configurable

**Comment by lemonzi** [comment]
I've merged the head of #39 into this branch, in preparation for when it's merged with master.

**Pull request closed without merging**